main/utils/textEntry.h

```diff
@@ -0,0 +1,143 @@
+/*! \file textEntry.h
+ *
+ * \section textEntry_design Design Philosophy
+ *
+ * The text entry utility provides a minimal interface for editing a single line of text of unlimited length.
+ * It can easily be adapted to any interface
+ *
+ * \section textEntry_usage Usage
```

Comment:
A little sparse

Reply:
Woops. That reminds me I need to add control displays here too...